src/trace/trace.c

```diff
@@ -96,6 +97,91 @@ static inline void mtrace_event(const char *data, uint32_t length)
 }
 #endif /* CONFIG_TRACEM */
 
+static int trace_filter_update_global(int32_t uuid_id, int32_t log_level)
+{
+	extern void *_trace_ctx_start;
+	extern void *_trace_ctx_end;
+	struct tr_ctx *ptr = (struct tr_ctx *)&_trace_ctx_start;
+	struct tr_ctx *end = (struct tr_ctx *)&_trace_ctx_end;
+	int cnt = 0;
+
+	while (ptr < end) {
+		if (ptr->uuid_p == uuid_id) {
+			ptr->level = log_level;
+			return 1;
+		}
+		if (ptr->uuid_p == -1) {
+			ptr->level = log_level;
+			++cnt;
+		}
+		++ptr;
+	}
+
+	return cnt;
+}
+
+static struct tr_ctx *trace_filter_ipc_comp_context(struct ipc_comp_dev *icd)
+{
+	switch (icd->type) {
+	case COMP_TYPE_COMPONENT:
+		return &icd->cd->tctx;
+	case COMP_TYPE_BUFFER:
+		return &icd->cb->tctx;
+	case COMP_TYPE_PIPELINE:
+		return &icd->pipeline->tctx;
+	}
+	/* each COMP_TYPE must be specified */
+	assert(NULL);
```

Comment:
Is this really fatal ?

Reply:
assertion replaced with log error and return NULL